Build error when building for Android with `disable_3d=yes`

### Tested versions

- Reproducible in: 4.4-stable
- Not reproducible in 4.3-stable and lower

### System information

Arch Linux x86_64 6.13.5-arch1-1

### Issue description

Godot doesn't compile when using the `disable_3d=yes` option in scons

```
platform/android/java_godot_lib_jni.cpp:478:28: error: use of undeclared identifier 'RenderingServer'; did you mean 'RenderingDevice'?
        String rendering_driver = RenderingServer::get_singleton()->get_current_rendering_driver_name();
                                  ^~~~~~~~~~~~~~~
                                  RenderingDevice
platform/android/display_server_android.h:38:7: note: 'RenderingDevice' declared here
class RenderingDevice;
      ^
platform/android/java_godot_lib_jni.cpp:478:28: error: incomplete type 'RenderingDevice' named in nested name specifier
        String rendering_driver = RenderingServer::get_singleton()->get_current_rendering_driver_name();
                                  ^~~~~~~~~~~~~~~~~
platform/android/display_server_android.h:38:7: note: forward declaration of 'RenderingDevice'
class RenderingDevice;
      ^
platform/android/java_godot_lib_jni.cpp:479:28: error: use of undeclared identifier 'RenderingServer'; did you mean 'RenderingDevice'?
        String rendering_method = RenderingServer::get_singleton()->get_current_rendering_method();
                                  ^~~~~~~~~~~~~~~
                                  RenderingDevice
platform/android/display_server_android.h:38:7: note: 'RenderingDevice' declared here
class RenderingDevice;
      ^
platform/android/java_godot_lib_jni.cpp:479:28: error: incomplete type 'RenderingDevice' named in nested name specifier
        String rendering_method = RenderingServer::get_singleton()->get_current_rendering_method();
                                  ^~~~~~~~~~~~~~~~~
platform/android/display_server_android.h:38:7: note: forward declaration of 'RenderingDevice'
class RenderingDevice;
      ^
```

Tested in Arch Linux and Ubuntu 20.04. No difference.
This didn't happen on 4.3 version
After some research it seems like this is connected to this commit: [28d1dcc](https://github.com/godotengine/godot/commit/28d1dccf6370a1754c9bd49b9e5ec3f15db1e535)

### Steps to reproduce

1. Download Godot source code and required build tools
2. Compile the engine with command `scons platform=android arch=x86_64 disable_3d=yes`

### Minimal reproduction project (MRP)

n/a

--- a/platform/android/java_godot_lib_jni.cpp
+++ b/platform/android/java_godot_lib_jni.cpp
@@ -49,6 +49,7 @@
 #include "core/config/project_settings.h"
 #include "core/input/input.h"
 #include "main/main.h"
+#include "servers/rendering_server.h"
 
 #ifndef _3D_DISABLED
 #include "servers/xr_server.h"

